dashboard/modules/metrics/metrics_head.py

```diff
@@ -23,14 +31,76 @@
 )
 
 
+class TaskProgress(BaseModel):
+    num_finished: int = 0
+    num_running: int = 0
+    nun_scheduled: int = 0
+    num_waiting_for_dependencies: int = 0
+    num_waiting_for_execution: int = 0
+
+
+prometheus_metric_map = {
+    "FINISHED": "num_finished",
+    "RUNNING": "num_running",
+    "SCHEDULED": "nun_scheduled",
+    "WAITING_FOR_DEPENDENCIES": "num_waiting_for_dependencies",
+    "WAITING_FOR_EXECUTION": "num_waiting_for_execution",
+}
+
+
 class MetricsHead(dashboard_utils.DashboardHeadModule):
-    def __init__(self, dashboard_head):
+    def __init__(
+        self, dashboard_head, http_session: Optional[aiohttp.ClientSession] = None
+    ):
         super().__init__(dashboard_head)
+        self.http_session = http_session or aiohttp.ClientSession()
+        self.prometheus_host = os.environ.get(
+            PROMETHEUS_HOST_ENV_VAR, DEFAULT_PROMETHEUS_HOST
+        )
         default_metrics_root = os.path.join(self._dashboard_head.session_dir, "metrics")
         self.metrics_root = os.environ.get(
             METRICS_OUTPUT_ROOT_ENV_VAR, default_metrics_root
         )
 
+    @routes.get("/api/progress")
+    async def get_progress(self, req):
+        """
+        Fetches the progress of tasks by job id. If job_id is not provided,
+        then we will fetch the progress across all jobs.
+        """
+        job_id = req.query.get("job_id")
+        job_id_query = f'{{JobId="{job_id}"}}' if job_id else ""
+        query = f"sum(ray_tasks{job_id_query}) by (State)"
+        async with self.http_session.get(
+            f"{self.prometheus_host}/api/v1/query?query={quote(query)}"
```

Comment:
How do we find a host from the prod? I remember we use Cortext instead of the prometheus

Reply:
We still have prometheus running in the head node and cortex should still have a query api we can use.